### PR TITLE
Bug - 2766 - Fix Candidate Results on Initial Render

### LINKED ISSUE #2766: talent - search results bug
toggling the filter to narrow down and then toggling off does not return back the original number of candidates

https://recordit.co/2mwW9UKKTS

### PR DESCRIPTION
**Pull request by esizer**
Resolves #2766 

## Summary

This adds an effect to run the query after the initial values have been set up which allows us to get the proper results on the first render.

## Notes

This could be unreliable based on how we are storing form state using the router. I think the fix here, while it works, may not be the desired end goal.

We added some `sessionStorage` state to the profile forms and when we have the opportunity, we should migrate the search form to use the same approach which may dodge some of the issues we are having (not guaranteed but a hunch).

### CODE REVIEW

**Comment by vd1992** [comment]
- initial candidate count appears correct now, query passes in variables
- filter toggling now behaves as expected
- one instance of confusing behaviour found, see below 

No filters yields 40 candidates, but selecting all the classifications possible yields 50 candidates?

image

**Comment by vd1992** [comment]
> > No filters yields 40 candidates, but selecting all the classifications possible yields 50 candidates?
> 
> Eagle eye @vd1992 . I'm going to look into this and see if it is related and see if it can be patched in this PR. Thanks!
> 
> 🐦 - (closest they have to an eagle)

I'm establishing a reviewing style for myself, trying to break PRs 💪

**Comment by esizer** [comment]
> No filters yields 40 candidates, but selecting all the classifications possible yields 50 candidates?

Okay, I have narrowed it down. I think it is to do with the [PoolCandidate.php#L92](https://github.com/GCTC-NTGC/gc-digital-talent/blob/bc4278b8a2b75c4f829e875d91f054742ba5fc7a/api/app/Models/PoolCandidate.php#L92) where we call the `orFilterByClassificationToSalary` method.

I think it has to do with the [PoolCandidate.php#L167](https://github.com/GCTC-NTGC/gc-digital-talent/blob/bc4278b8a2b75c4f829e875d91f054742ba5fc7a/api/app/Models/PoolCandidate.php#L167) specifically but can't be certain.

I do know in laravel when you call `orWhere` it treats the result as an addition to existing query so what I think this produces is:

**Get candidates with these salary expectations OR candidates in these classifications and pool.**

Not sure if that makes sense or not 🤷

**Comment by vd1992** [comment]
@petertgiles 
please weigh in on the above

**Review by petertgiles — APPROVED** [review]
This works for me in testing.  I'm seeing the issue Vachan uncovered but I'm thinking that's out of scope for this.

File: frontend/talentsearch/src/js/components/search/SearchForm.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import React, { useMemo } from "react";
+import React, { useCallback, useMemo } from "react";
```
[the remark this answers is not in the document]
Comment (esizer) [reply frontend/talentsearch/src/js/components/search/SearchForm.tsx]: Fixed with [remove unused import](https://github.com/GCTC-NTGC/gc-digital-talent/pull/2778/commits/1448e20be75434cbff620820a98babb3e7712c19)

**Pull request merged**